src/initial_questions/inital_questions.py

```diff
@@ -28,24 +29,44 @@ def fetch_relevant_data_from_k8s_cluster(
         """Fetch the relevant data from Kubernetes cluster based on specified K8s resource in message."""
         ...
 
+    def apply_token_limit(self, text: str, token_limit: int) -> str:
+        """Reduces the amount of tokens of a string by truncating exeeding tokens.
+        Takes the template into account."""
+        ...
+
 
 class InitialQuestionsHandler:
     """Handler that generates initial questions."""
 
     chain: typing.Any
+    _model: IModel
+    _template: str
+
+    def __init__(self, model: IModel, template: str | None = None) -> None:
+        self._model = model
+        self._template = template or INITIAL_QUESTIONS_PROMPT
 
-    def __init__(
-        self,
-        model: IModel,
-    ) -> None:
-        prompt_template: str = INITIAL_QUESTIONS_PROMPT
         prompt = PromptTemplate(
-            template=prompt_template,
+            template=self._template,
             input_variables=["context"],
         )
         output_parser = QuestionOutputParser()
         self.chain = prompt | model.llm | output_parser
 
+    def apply_token_limit(self, text: str, token_limit: int) -> str:
+        """Reduces the amount of tokens of a string by truncating exeeding tokens.
+        Takes the template into account."""
+        tokenizer = tiktoken.encoding_for_model(self._model.name)
```

Comment:
can you cache the `tokenizer` instance as an object field as it is initialised for every initial questions call?

Reply:
Good idea.